herbert_core/utilities/maths/equal_to_tol.m

```diff
@@ -405,7 +405,8 @@
 
     % Check fieldnames are identical
     fields=fieldnames(a);
-    if ~isequal(fields,fieldnames(b))
+    fb = fieldnames(b);
+    if ~(isempty(fields)&&isempty(fb)) && ~any(ismember(fields,fb))
         ok=false;
```

Comment:
Would use this as an opportunity to use error failure rather than returning `ok`

Reply:
its used in number of places with ok check (including Horace) I would be highly scared to change all of these. Its also possible that this generic function is not creating errors if the objects are different but uses it in some logic. 

I suspect, changing it to throwing an error would bring much more work then one would like to accept.